core/controller/src/main/scala/org/apache/openwhisk/core/controller/Actions.scala

```diff
@@ -216,11 +224,23 @@ trait WhiskActionsApi extends WhiskCollectionAPI with PostActionActivation with
           case _ => entitlementProvider.check(user, content.exec)
         }
 
+        val latestDocId = WhiskActionVersionList.get(entityName, entityStore).map { result =>
```

Comment:
This seems to be reused everywhere. Can we make this a future function we can just call to get the latest doc id?

Reply:
yes, I will update